src/document/position.js

```diff
@@ -97,13 +119,13 @@ CKEDITOR.define( [ 'utils', 'ckeditorerror' ], ( utils, CKEditorError ) => {
 		}
 
 		/**
-		 * Two positions equal if paths are equal.
+		 * Two positions equal if paths are equal and roots are the same.
 		 *
 		 * @param {document.Position} otherPosition Position to compare.
 		 * @returns {Boolean} True if positions equal.
 		 */
 		isEqual( otherPosition ) {
```

Comment:
Well, you are right. It all boils down to code readability vs API maintenance. `if ( a.isEqual( b ) ) { }` for sure looks better than `if ( a.compareWith( b ) == Position.SAME )` and is easier to remember. Also when you try to understand part of code your brain has to process what compareWith Position SAME actually means, so I think that it is better to have function names that sounds natural and are not distracting. I like the idea of `Position.isBefore` and `Position.isAfter`. I know that on several occasions you fought for keeping API simple but the main reason for keeping it simple is easy maintenance and those methods would be very simple and will be based on just one method `compareWith`. I am all for such methods.